src/effects/nyquist/Nyquist.cpp

```diff
@@ -117,6 +117,68 @@ enum
 static const wxChar *KEY_Command = wxT("Command");
 static const wxChar *KEY_Parameters = wxT("Parameters");
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// NyquistEffectUIValidator
+//
+///////////////////////////////////////////////////////////////////////////////
+struct NyquistEffect::SharedValidatorState final
+{
+   explicit SharedValidatorState(NyquistEffect& effect)
+       : mEffect(&effect), mParent(mEffect->GetUIParent())
+   {
+      mParent->PushEventHandler(&effect);
+   }
+
+   ~SharedValidatorState()
+   {
+      Reset();
+   }
+
+   void Reset()
+   {
+      if (mEffect == nullptr)
```

Comment:
If the constructor takes a reference to the effect, how can mEffect by null?

Reply:
Because I need to ensure that Reset is only called once.

Reply:
Oh yes, not just in destruction.